lib/activerecord-import/import.rb

```diff
@@ -582,7 +582,9 @@ def find_associated_objects_for_import(associated_objects_by_class, model)
           child.public_send("#{association_reflection.foreign_key}=", model.id)
           # For polymorphic associations
           association_reflection.type.try do |type|
-            child.public_send("#{type}=", model.class.name)
+            model_class_name = model.class.superclass.name
+            model_class_name = model.class.name if ["ActiveRecord::Base", "ApplicationRecord"].include? model_class_name
```

Comment:
I think this can be simplified by just calling:

`child.public_send("#{type}=", model.base_class.name)`